src/zc/buildout/easy_install.py

```diff
@@ -399,11 +400,11 @@ def _call_easy_install(self, spec, ws, dest, dist):
             for d in dists:
                 newloc = os.path.join(dest, os.path.basename(d.location))
                 if os.path.exists(newloc):
-                    if os.path.isdir(newloc):
-                        shutil.rmtree(newloc)
-                    else:
-                        os.remove(newloc)
-                os.rename(d.location, newloc)
+                    logger.warn(
+                        "Path %s already exists.\nIf this contains a wrong "
+                        "package, please remove it yourself.", newloc)
+                else:
+                    os.rename(d.location, newloc)
```

Comment:
Speaking of non-atomic ops, the function `_get_dist()` below calls `shutil.copytree()` directly into the destination when the source of a dist is an egg directory.

This is a rare situation (when was the last time you saw an unpacked **egg** outside of the buildout be used as the `find-links` of a package in the buildout?), but if it happens, one buildout could see that an egg exists and attempt to use it before another buildout is finished copying it.

I suggest the `copytree()` be done in a `tempdir()` inside the egg directory and then `os.rename()`d like `_call_easy_install()` does here.